src/SFA.DAS.LevyTransferMatching.Web/Controllers/OpportunitiesController.cs

```diff
@@ -2,24 +2,69 @@
 using System.Threading.Tasks;
 using Microsoft.AspNetCore.Authorization;
 using SFA.DAS.LevyTransferMatching.Web.Orchestrators;
+using SFA.DAS.LevyTransferMatching.Web.Models.Opportunities;
+using System.Data;
+using SFA.DAS.Authorization.Mvc.Attributes;
 
 namespace SFA.DAS.LevyTransferMatching.Web.Controllers
 {
-    [AllowAnonymous]
-    [Route("opportunities")]
     public class OpportunitiesController : Controller
     {
-        private readonly IOpportunitiesOrchestrator _searchFundingOrchestrator;
+        private readonly IOpportunitiesOrchestrator _opportunitiesOrchestrator;
 
         public OpportunitiesController(IOpportunitiesOrchestrator searchFundingOrchestrator)
         {
-            _searchFundingOrchestrator = searchFundingOrchestrator;
+            _opportunitiesOrchestrator = searchFundingOrchestrator;
         }
 
         public async Task<IActionResult> Index()
         {
-            var viewModel = await _searchFundingOrchestrator.GetIndexViewModel();
+            var viewModel = await _opportunitiesOrchestrator.GetIndexViewModel();
             return View(viewModel);
         }
+
+        [Route("opportunities/{encodedId}")]
```

Comment:
We should probably name the parameter "encodedPledgeId", as that is clearer (we'll likely end up referencing this in authorisation handlers too, not just here).

Reply:
The idea was that `encodedId` was within the context of the entity opportunities. I don't see an issue with it personally (as on the authorisation handlers, you could still use `encodedPledgeId`), but I've changed this as requested.